packages/odyssey-react-mui/src/ThemeProvider.tsx

```diff
@@ -10,11 +10,34 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
+import createCache from "@emotion/cache";
+import { CacheProvider } from "@emotion/react";
 import { ThemeProvider as MuiThemeProvider } from "@mui/material/styles";
-import { ReactElement } from "react";
+import { memo, ReactElement, useMemo } from "react";
 
 import { odysseyTheme } from "./theme";
+import { useUniqueAlphabeticalId } from "./useUniqueAlphabeticalId";
 
-export const ThemeProvider = ({ children }: { children: ReactElement }) => (
-  <MuiThemeProvider theme={odysseyTheme}>{children}</MuiThemeProvider>
-);
+const ThemeProvider = ({ children }: { children: ReactElement }) => {
+  const uniqueAlphabeticalId = useUniqueAlphabeticalId();
+
+  const emotionCache = useMemo(
+    () =>
+      createCache({
+        key: uniqueAlphabeticalId,
+        // @ts-expect-error ts(2345)
+        nonce: window.cspNonce,
```

Comment:
Can we expose a way for the application to pass the nonce as opposed to assuming it is on window.cspNonce.

Reply:
f264b50d5c737fcaa672eadb4d3302acfd136582